compiler/rustc_mir_transform/src/check_unsafety.rs

```diff
@@ -134,6 +134,12 @@ impl<'tcx> Visitor<'tcx> for UnsafetyChecker<'_, 'tcx> {
                     self.register_violations(violations, used_unsafe_blocks.items().copied());
                 }
             },
+            Rvalue::Cast(CastKind::Transmute, ..) => {
+                self.require_unsafe(
+                    UnsafetyViolationKind::General,
+                    UnsafetyViolationDetails::CallToUnsafeFunction,
+                );
+            }
```

Comment:
Should I just delete the section, then, rather than `bug`ging it?  Like how `visit_statement` in this visitor doesn't check for `CopyNonOverlapping` in `StatementKind::Intrinsic`?